### PR TITLE
clean our repo before running CI

### PR DESCRIPTION
**Pull request by barcharcraz**
# Description

tell azure pipelines to clean our repo before building, to prevent badness and broken CI pipelines.

This is a workaround for #396

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [x] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

File: azure-devops/run_build.yml [hunk 1]
```diff
@@ -2,74 +2,75 @@
...
-    - task: run-cmake@0
-      displayName: 'Build the STL'
-      timeoutInMinutes: 10
-      inputs:
-        cmakeListsTxtPath: 'CMakeSettings.json'
-        useVcpkgToolchainFile: true
-        configurationRegexFilter: '.*${{ parameters.targetPlatform }}.*'
-        buildDirectory: $(Build.ArtifactStagingDirectory)/${{ parameters.targetPlatform }}
+    variables:
+      vcpkgLocation: "$(Build.SourcesDirectory)/vcpkg"
```
Comment (CaseyCarter) [inline azure-devops/run_build.yml]: What's with all of these "single-quote to double-quote" changes? They greatly obscure the actual functional changes. Am I correct that the addition of `clean: true` on line 13 is the only functional change?

Comment (barcharcraz) [reply azure-devops/run_build.yml]: ugh it's the red-hat yaml extension for vscode.

Comment (barcharcraz) [reply azure-devops/run_build.yml]: trying to be a little too clever.

**Comment by StephanTLavavej** [comment]
I've linked #396, but please mention it in your commit message.

**Pull request merged**